File: 5/5-298.pas
// Автор В.Н. Шубинкин

uses School;

function R(N: integer): integer;
begin
  var N_bi := Bin(N);
  loop 2 do
  begin
    var c1 := N_bi.CountOf('1');
    if c1 mod 2 = 0 then N_bi := N_bi[2:]
    else N_bi := '1'*(c1 + 1)
  end;
  result := Dec(N_bi, 2)
end;


begin
var c := 0;
for var n := 1 to 1000 do c += ord(R(n) = 7);
print(c)
end.

File: Dima/5/5-298.py
# Автор В.Н. Шубинкин

def r(n):
    n_bi = bin(n)[2:]
    for _ in range(2):
        c1 = n_bi.count('1')
        if c1 % 2:
            n_bi = '1'*(c1 + 1)
        else:
            n_bi = n_bi[1:]
    return int(n_bi, 2)

c = 0
for n in range(1, 1001):
    c += r(n) == 7
print(c)
